modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/LayoutProvider.es.js

```diff
@@ -322,10 +324,48 @@ class LayoutProvider extends Component {
 						successPageSettings,
 					},
 				}))}
+				<ClayModal
+					body={Liferay.Language.get(
+						'a-rule-is-applied-to-this-field'
+					)}
+					elementClasses={'lfr-ddm-forms-delete-rule'}
+					events={{
+						clickButton: this._handleDeleteFieldModalButtonClicked.bind(
+							this
+						),
+					}}
+					footerButtons={[
+						{
+							alignment: 'right',
+							label: Liferay.Language.get('cancel'),
+							style: 'secondary',
+							type: 'close',
+						},
+						{
+							alignment: 'right',
+							label: Liferay.Language.get('delete'),
+							style: 'primary',
+							type: 'button',
+						},
+					]}
+					ref={'existingRuleModal'}
+					size={'lg'}
+					title={Liferay.Language.get(
+						'delete-field-with-rule-applied'
+					)}
+				/>
 			</span>
 		);
 	}
 
+	_handleDeleteFieldModalButtonClicked(event) {
+		const {activePage, fieldName} = this.refs.existingRuleModal.data;
+
+		if (event.target.classList.contains('btn-primary')) {
```

Comment:
@alinedoleron, would be better check for the button's label? Check only for the class `btn-primary` isn't so intuitive for the code reader and also will require a rework from our side if this class changes.

Reply:
I have a question. Should the button labels be localisable? I thought when we changed the locale on `Control-panel > Instance settings > Localization` the label should be according to the language. That was the reason I didn't compare with the label.

Reply:
Yes. Now, I see your point, @alinedoleron. Does this button has a unique identifier (`id`) which we can use? Using this strategy would make this code stronger against CSS/UI changes. 🙂

Reply:
@natocesarrego, actually, after talking to Matu right now, he said the first solution I has done is the right. He said also there's no way to put a unique identifier on the footersButtons. I asked him about the class changes, but he said it's very hard this to be changed, as it's a pattern on the web. As an alternative, he said I could try to verify if the modal had close after the click, but it didn't work. For both cases after clicking on the buttons the modal keeps opened before the verification. So, I'll change it back to the code before.